verifiable-db/src/query/batching/row_chunk/aggregate_chunks.rs

```diff
@@ -0,0 +1,765 @@
+            predecessor_info: NeighborInfo::new(
+                node_C.value,
+                Some(HashOut::from_bytes((&node_C_hash).into())),
+            ),
+            successor_info: NeighborInfo::new_dummy_successor(),
+        };
+        let (expected_outputs, expected_num_overflows) = {
+            let outputs = [first_chunk_outputs.clone(), second_chunk_outputs.clone()];
+            let mut num_overflows = fist_chunk_num_overflows + second_chunk_num_overflows;
+            let outputs = ops
+                .into_iter()
+                .enumerate()
+                .flat_map(|(i, op)| {
+                    let (out, overflows) = aggregate_output_values(i, &outputs, op);
+                    num_overflows += F::from_canonical_u32(overflows);
+                    out
+                })
+                .collect_vec();
+            (
+                OutputValues::from_fields(&outputs),
+                num_overflows.to_canonical_u64(),
+            )
+        };
+        let expected_count = (first_chunk_count + second_chunk_count).to_canonical_u64();
+
+        let expected_chunk = RowChunkData::<MAX_NUM_RESULTS> {
+            left_boundary_row: BoundaryRowData {
+                row_node_info: node_E_info.clone(),
+                index_node_info: index_node_info.clone(),
+            },
+            right_boundary_row: BoundaryRowData {
+                row_node_info: node_G_info,
+                index_node_info: index_node_info.clone(),
+            },
+            chunk_tree_hash: root,
+            output_values: expected_outputs.clone(),
+            num_overflows: expected_num_overflows,
+            count: expected_count,
+        };
+
+        assert_eq!(proof.public_inputs, expected_chunk.to_fields());
+
+        // test with second chunk being dummy; we use a non-consecutive chunk as the dummy one: the row chunk
+        // given by node_G only
+        let second_chunk = RowChunkDataInput {
+            left_boundary_row_path: merkle_path_inputs_G,
+            left_boundary_row_node: node_G.clone(),
+            left_boundary_index_path: index_node_merkle_path,
+            left_boundary_index_node: index_node.clone(),
+            right_boundary_row_path: merkle_path_inputs_G.clone(),
+            right_boundary_row_node: node_G.clone(),
+            right_boundary_index_path: index_node_merkle_path,
+            right_boundary_index_node: index_node.clone(),
+            chunk_count: second_chunk_count,
+            chunk_num_overflows: second_chunk_num_overflows,
+            chunk_output_values: second_chunk_outputs.clone(),
+        };
+        let circuit = TestAggregateChunks {
+            first: first_chunk.clone(),
+            second: second_chunk.clone(),
+            min_primary: None,
+            max_primary: None,
+            min_secondary: None,
+            max_secondary: None,
+            primary_index_id,
+            secondary_index_id,
+            ops,
+            is_second_dummy: true,
```

Comment:
lol it took me a while to understand that second test case until I see that flag. Can you write the comment here please, because i got confused about node G, i thought it should be the right boundary but then i saw that flag.

Reply:
Added a comment specifying that we set the chunk as dummy in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/396/commits/5787dc75a5d6df1a467910e6fc2b524e21943460